federation-2/build_for_local_dev.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+set -euo pipefail
+trap 'echo "ERR (rc: $?)"' ERR
+
+# Creates a new `supergraph` binary for use in local development and copies it to the local ~/.rover/bin directory for
+# end-to-end testing with `rover` and Apollo Workbench.
+
+# Get the path to the local federation repo
+if [ -z "${1:-}" ]; then
+  echo "Usage: $0 <path-to-federation-repo>"
+  exit 1
+fi
+FEDERATION_JS_PATH="$1"
+
+# Stage 1: Build a local copy of all the JS federation stuff
+pushd "$FEDERATION_JS_PATH"
+
+npm run compile
+
+pushd internals-js
+INTERNALS_TARBALL=$(npm pack | tail -n 1)
+VERSION=$(echo "$INTERNALS_TARBALL" | grep -oE '\d+\.\d+\.\d+(-\w+\.\d+)?')
+STABLE_COMPONENT=$(echo "$VERSION" | cut -d'-' -f1)
+popd
+
+pushd query-graphs-js
+# Use the packed version of internals
+npm install "$FEDERATION_JS_PATH"/internals-js/"$INTERNALS_TARBALL"
+QUERY_GRAPHS_TARBALL=$(npm pack | tail -n 1)
+# Restore original install
+npm i -E @apollo/federation-internals@"$VERSION"
```

Comment:
I'd prefer running `npm i -w <project>` from the workspace root. In its current form, you're going to see a lot more nested `node_modules` per project and the top-level lockfile won't be used or updated.